app/js/RangesTracker.js

```diff
@@ -728,8 +730,10 @@ const load = function () {
           return result
         } else if (a.i != null && b.d != null) {
           return 1
-        } else {
+        } else if (a.d != null && b.i != null) {
           return -1
+        } else {
+          return 0
```

Comment:
Logic seems correct to me (and necessary due to the change to Array.sort I imagine...)

Did the tests fail before making this change out of curiosity? I'm just wondering how easy it might be for someone to accidentally break something, or if it would be helpful to have a small comment to say that it must return a number as a warning to future developers.

Reply:
The tests passed before. And I didn't find a case where the sort function misbehaved. It's possible that the comparator never caused problems. It's still not well defined. You can build two objects, `a` and `b`, such that `cmp(a,b) == -1` and `cmp(b,a) == -1`, which would mean that both `a < b` and `b < a`.

So, it's not a necessary change triggered by an actual problem, but it's a logic problem that has the potential to trigger undefined behaviour.